mantidproject/mantid
# Add run number to output of SaveNXSPE

### Description of work

The run number of the input workspace is now added to the instrument section .nxspe files. 

I have also updated the nxspe file version number.

Fixes #37654.

**Report to:** @abuts 

### To test:

So far, I have ensured that an .nxspe file saved with the updated version of SaveNXSPE can still be opened with DAVE - MSlice. 

To verify that the run number is set as expected, I have used [HDFView](https://www.hdfgroup.org/downloads/hdfview/).

It would be good to check if Horace can use the run number written to the .nxspe file.

from @abuts 
All these checks have been done and are successful. In addition to that, added code which correctly identifies energies of detectors for indirect instrument and stores them accordingly.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
